stdr_server/src/stdr_server.cpp

```diff
@@ -682,5 +752,126 @@ namespace stdr_server {
+    // Any marker sent with the same namespace and id will overwrite the old one
+    marker.ns = "co2";
+	marker.id = atoi(msg.id.c_str());
+	
+    // Set the color specific to the source type -- be sure to set alpha to something non-zero!
+    marker.color.r = 0.0f;
+    marker.color.g = 1.0f;
+    marker.color.b = 0.0f;
+    marker.color.a = 1.0;
+    
+    return marker;
+  }
+  
+  /**
+  @brief Translate the stdr_ThermalSource message into a marker message
+  **/
+  visualization_msgs::Marker Server::toMarker(const stdr_msgs::ThermalSource& msg)
+  {
+	visualization_msgs::Marker marker = createMarker(msg);
+	
+	// Set the namespace and id for this marker.  This serves to create a unique ID
+    // Any marker sent with the same namespace and id will overwrite the old one
+    marker.ns = "thermal";
+	marker.id = atoi(msg.id.c_str());
+	
+    // Set the color specific to the source type -- be sure to set alpha to something non-zero!
+    marker.color.r = 1.0f;
+    marker.color.g = 0.0f;
+    marker.color.b = 0.0f;
+    marker.color.a = 1.0;
+    
+    return marker;
+  }
+  
+  /**
+  @brief Translate the stdr_SoundSource message into a marker message
+  **/
+  visualization_msgs::Marker Server::toMarker(const stdr_msgs::SoundSource& msg)
+  {
+	visualization_msgs::Marker marker = createMarker(msg);
+	
+	// Set the namespace and id for this marker.  This serves to create a unique ID
+    // Any marker sent with the same namespace and id will overwrite the old one
+    marker.ns = "sound";
+	marker.id = atoi(msg.id.c_str());
+	
+    // Set the color specific to the source type -- be sure to set alpha to something non-zero!
+    marker.color.r = 0.0f;
+    marker.color.g = 0.0f;
+    marker.color.b = 1.0f;
+    marker.color.a = 1.0;
+    
+    return marker;
+  }
+  
+  /**
+  @brief Translate the stdr_SoundSource message into a marker message
+  **/
+  visualization_msgs::Marker Server::toMarker(const stdr_msgs::RfidTag& msg)
+  {
+    visualization_msgs::Marker marker = createMarker(msg);
+	
+    // Set the namespace and id for this marker.  This serves to create a unique ID
+    // Any marker sent with the same namespace and id will overwrite the old one
+    marker.ns = "rfid";
+    marker.id = atoi(msg.tag_id.c_str());
+	
+    // Set the color specific to the source type -- be sure to set alpha to something non-zero!
+    marker.color.r = 0.0f;
+    marker.color.g = 0.0f;
+    marker.color.b = 0.0f;
+    marker.color.a = 1.0;
+    
+    return marker;
+  }
+  
+    
+  /**
+  @brief Creates a marker message corresponding to every element of msg that is 
+  independent of the source's specific type 
+  **/
+  template <class SourceMsg> visualization_msgs::Marker Server::createMarker(const SourceMsg& msg)
+  {
+    visualization_msgs::Marker marker;
+    // Set the frame ID and timestamp.  See the TF tutorials for information on these.
+    marker.header.frame_id = "/world";
+    marker.header.stamp = ros::Time::now();
```

Comment:
You should use `ros::Time()` instead, making the marker independent of the creation time, when it comes to drawing.